src/sgp/Smack_Stub.cc

```diff
@@ -130,27 +144,23 @@ Smack* SmackOpen(SGPFile* FileHandle, UINT32 Flags, UINT32 ExtraFlag)
 	SmackCheckStatus(smkstatus);
 	smkstatus = smk_info_audio(flickinfo->Smacker, NULL, a_channels, a_depth, a_rate);
 	SmackCheckStatus(smkstatus);
+	//SmackPrintFlickInfo(width, height, scale, framecount, usf, a_channels[SMKTRACK], a_depth[SMKTRACK], a_rate[SMKTRACK]);
 
 	SmkVideoSwitch(flickinfo->Smacker, DISABLE);
 
 	flickinfo->Frames=framecount;
 	flickinfo->Height=height;
 	flickinfo->Width=width;
 	flickinfo->FramesPerSecond = usf;
-	// calculated audio memory for downsampling 44100->22050
-	audiosamples = ((flickinfo->Frames / (usf/1000)) * (a_rate[SMKTRACK]) * 16 *  a_channels[SMKTRACK]);
-	audiobuffer = (INT16*) malloc(audiosamples);
-	if (!audiobuffer)
+	audiobuffer = NULL;
+	audiolen = SmackGetAudio(flickinfo->Smacker, &audiobuffer);
+	if (audiobuffer != NULL);
 	{
-		free(flickinfo);
-		return NULL;
+		//SmackWriteAudio( audiobuffer, audiolen); // are getting right audio data?
+		// shoot and forget... audiobuffer should be freed by SoundMan
```

Comment:
This comment says that audiobuffer should be freed by Soundman but it is freed two lines below. Is this intended?

Reply:
It's an old comment that I didn't notice, thanks.